src/Data/List/Extrema/Core.agda

```diff
@@ -0,0 +1,120 @@
+------------------------------------------------------------------------
+-- The Agda standard library
+--
+-- Core lemmas needed to make list argmin/max functions work
+------------------------------------------------------------------------
+
+{-# OPTIONS --without-K --safe #-}
+
+open import Relation.Binary using (Trans; TotalOrder; Setoid)
+
+module Data.List.Extrema.Core
+  {b ℓ₁ ℓ₂} (totalOrder : TotalOrder b ℓ₁ ℓ₂) where
+
+open import Algebra.FunctionProperties
+import Algebra.Construct.NaturalChoice.Min as Min
+import Algebra.Construct.NaturalChoice.Max as Max
+open import Data.Product using (_×_; _,_)
+open import Data.Sum using (_⊎_; inj₁; inj₂)
+open import Level using (Level)
+open import Relation.Binary.PropositionalEquality using (_≡_)
+
+open import Algebra.Construct.LiftedChoice
+-- import RoutingLib.Relation.Binary.Construct.NonStrictToStrict.TotalOrder as NonStrictToStrict
+
+open TotalOrder totalOrder renaming (Carrier to B)
+open import Relation.Binary.Construct.NonStrictToStrict _≈_ _≤_
+  using (_<_; <-≤-trans; ≤-<-trans)
+
+------------------------------------------------------------------------
+-- Setup
+
+-- open NonStrictToStrict totalOrder using (_<_; ≤-<-trans; <-≤-trans)
+
+open Max totalOrder
+open Min totalOrder
+
+private
+
+  variable
+    a : Level
+    A : Set a
+
+  <-transʳ : Trans _≤_ _<_ _<_
```

Comment:
It seems to me like the next 3 functions and 4 lemmas belong in `Relation.Binary.Construct.NonStrictToStrict` (or a `.Properties` sub-module).

Reply:
So the proof of the two functions are indeed from `Relation.Binary.Construct.NonStrictToStrict`. They can't be instantiated further as that module doesn't actually know the order is a partial order. I have toyed in my own library with
adding `Relation.Binary.Construct.NonStrictToStrict.PartialOrder` with these proofs instantiated...

Reply:
Just had a look at these again, and I've changed my mind. The first two functions can't be generalised without creating a new module as stated above. The 4 lemmas have to depend on `f` in order to be of the right shape, so even if you strip `f` out and add the lemmas to `Algebra.Construct.NaturalChoice.Min/Max` then you still basically have to re-create the whole function in order to add `f` back in again.